src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1644,76 +1644,105 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
 
         private void MakeSeparatorListVectorized(ref ValueListBuilder<int> sepListBuilder, char c, char c2, char c3)
         {
-            ref ushort source = ref Unsafe.As<char, ushort>(ref _firstChar);
-
             nuint offset = 0;
             nuint lengthToExamine = (nuint)(uint)Length;
 
-            if (Vector256.IsHardwareAccelerated && lengthToExamine >= (nuint)Vector256<ushort>.Count)
-            {
-                Vector256<ushort> v1 = Vector256.Create((ushort)c);
-                Vector256<ushort> v2 = Vector256.Create((ushort)c2);
-                Vector256<ushort> v3 = Vector256.Create((ushort)c3);
+            ref ushort source = ref Unsafe.As<char, ushort>(ref _firstChar);
 
-                while (offset <= lengthToExamine - (nuint)Vector256<ushort>.Count)
+            if (Vector256.IsHardwareAccelerated)
+            {
+                if (lengthToExamine >= (nuint)Vector256<ushort>.Count)
                 {
-                    Vector256<ushort> vector = Vector256.LoadUnsafe(ref source, offset);
-                    Vector256<ushort> vector1 = Vector256.Equals(vector, v1);
-                    Vector256<ushort> vector2 = Vector256.Equals(vector, v2);
-                    Vector256<ushort> vector3 = Vector256.Equals(vector, v3);
-                    Vector256<byte> cmp = (vector1 | vector2 | vector3).AsByte();
+                    Vector256<ushort> v1 = Vector256.Create((ushort)c);
+                    Vector256<ushort> v2 = Vector256.Create((ushort)c2);
+                    Vector256<ushort> v3 = Vector256.Create((ushort)c3);
```

Comment:
Unrelated to this PR, am just curios if our guidelines allow to use `var` here, the type of vector should be pretty obvious from the expression on the right.

Reply:
The guidelines say var should only be used for a ctor or explicit cast.  While it's arguable that Create is equivalent to a ctor, there's nothing that requires it to return the same type it's declared on, and in fact there are cases where Create methods don't, e.g. `File.Create`.